design-document.md

```diff
@@ -14,4 +24,210 @@
 
 ## Open questions and system parameters
 
-## Cryptographic dependencies
+## Cryptographic components
+
+### The input-validation protocol
+
+**Notation.**
+Fix a finite field K. We let K^n denote the set of vectors over K of length n.
+We write the linear, s-way secret sharing of an element x of K^n as {x:1}, ...,
+{x:s}, where {x:i} is the share held by the i-th party. We write {x} as
```

Comment:
I think it's worth clarifying that `{x:i}` is itself an element of `K^n` (right?)

Reply:
`{x:i}` is a share of `x`, so it's just an element of `K`.

Reply:
Yes, done.